### PR TITLE
Base image

### PR DESCRIPTION
**Pull request by babsey**
This PR adds Dockerfiles for two base-images for build and deployment stages.

### CODE REVIEW

File: src/3.x/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# BUILD
+
+FROM docker-registry.ebrains.eu/nest/nest-simulator-build-base as builder
+LABEL maintainer="s.graber@fz-juelich.de"
+
+ARG NEST_VERSION=3.1
+    SRC_PATH=/tmp/nest-simulator
```
Comment (steffengraber) [inline src/3.x/Dockerfile:7]: There should be `ARG ` in front of the argument.

**Review by steffengraber — CHANGES_REQUESTED** [review]
After all the deploy image started with "nest-server" raised the `error: /usr/local/bin/entrypoint.sh: line 38: exec: uwsgi: not found`

**Comment by steffengraber** [comment]
The messoge now:

    tar: /tmp/nest-simulator-3.1: Cannot open: No such file or directory
    tar: Error is not recoverable: exiting now

I think to prepare NEST 3.2 it would be better to start from the dev version and not v3.1. What do you think?

**Comment by terhorstd** [comment]
> I think to prepare NEST 3.2 it would be better to start from the dev version and not v3.1. What do you think?

The images should both (all) be built, since both have a different purpose. The releases have to trigger a build, obviously, but also the intermediate versions should be available to developers. The intermediate images need not be retained long term.

**Comment by babsey** [comment]
It is the first draft to built it on GitLab CI and deploy on docker-registry.ebrains.

Since there is no release of v3.2 existed, I took v3.1 as example for Dockerfiles templates. 
Sure you can change it.

**Comment by steffengraber** [comment]
I would see the goal of this PR as how the separation of the current image into build and deploy can work. The development of the right CI can be done in another PR.

**Review by steffengraber — APPROVED** [review]
Very good! Thank you.

**Pull request merged**